File: src/01-easy/ascii-art/solution.pl
chomp(my $L=<STDIN>); chomp(my $H=<STDIN>); chomp(my $T=uc <STDIN>);
my @rows; for(1..$H){chomp(my $r=<STDIN>);push @rows,$r}
for my $i(0..$H-1){
    for my $c(split//,$T){
        my $idx=ord($c)-65; $idx=26 if $idx<0||$idx>25;
        print substr($rows[$i],$idx*$L,$L);
    }
    print "\n";
}

File: src/01-easy/ascii-art/solution.php
<?php
$L=intval(fgets(STDIN));$H=intval(fgets(STDIN));$T=strtoupper(trim(fgets(STDIN)));
$rows=[];for($i=0;$i<$H;$i++)$rows[]=rtrim(fgets(STDIN),"\n");
for($i=0;$i<$H;$i++){
    $line="";
    for($j=0;$j<strlen($T);$j++){
        $idx=ord($T[$j])-65;if($idx<0||$idx>25)$idx=26;
        $line.=substr($rows[$i],$idx*$L,$L);
    }
    echo $line."\n";
}
